win32_event_log/datadog_checks/win32_event_log/win32_event_log.py

```diff
@@ -1,300 +1,24 @@
-                )
-
-                # Since WQL only compares on the date and NOT the time, we have to
-                # do a secondary check to make sure events are after the last
-                # timestamp
-                if log_ev.is_after(last_ts):
-                    self.event(log_ev.to_event_dict())
-                else:
-                    self.log.debug('Skipping event after %s. ts=%s', last_ts, log_ev.timestamp)
-
-            # Update the last time checked
-            self.last_ts[instance_key] = datetime.utcnow()
-
-    def _dt_to_wmi(self, dt):
-        ''' A wrapper around wmi.from_time to get a WMI-formatted time from a
-            time struct.
-        '''
-        return from_time(
-            year=dt.year,
-            month=dt.month,
-            day=dt.day,
-            hours=dt.hour,
-            minutes=dt.minute,
-            seconds=dt.second,
-            microseconds=0,
-            timezone=0,
-        )
-
-
-class LogEvent(object):
-    def __init__(self, ev, log, hostname, tags, notify_list, tag_event_id, event_format, event_priority):
-        self.event = self._normalize_event(ev.copy())
-        self.log = log
-        self.hostname = hostname
-        self.tags = self._tags(tags, self.event['EventCode']) if tag_event_id else tags
-        self.notify_list = notify_list
-        self.timestamp = self._wmi_to_ts(self.event['TimeGenerated'])
-        self._format = event_format
-        self.event_priority = event_priority
-
-    @staticmethod
-    def _normalize_event(event):
-        for field in INTEGER_PROPERTIES:
-            if field in event:
-                event[field] = int(event[field])
-        return event
-
-    @property
-    def _msg_title(self):
-        return '{logfile}/{source}'.format(logfile=self.event['Logfile'], source=self.event['SourceName'])
-
-    @property
-    def _msg_text(self):
-        """
-        Generate the event's body to send to Datadog.
-
-        Consider `event_format` parameter:
-        * Only use the specified list of event properties.
-        * If unspecified, default to the EventLog's `Message` or `InsertionStrings`.
-        """
-        msg_text = u""
-
-        if self._format:
-            msg_text_fields = ["%%%\n```"]
-
-            for event_property in self._format:
-                property_value = self.event.get(event_property)
-                if property_value is None:
-                    self.log.warning(u"Unrecognized `%s` event property.", event_property)
-                    continue
-                msg_text_fields.append(
-                    u"{property_name}: {property_value}".format(
-                        property_name=event_property, property_value=property_value
-                    )
-                )
-
-            msg_text_fields.append("```\n%%%")
-
-            msg_text = u"\n".join(msg_text_fields)
+        if is_affirmative(instance.get('use_wmi', False)):
+            return super(Win32EventLogCheck, cls).__new__(cls)
```

Comment:
By moving the code to `legacy.py` we will lose the commit history. I think we should avoid that when possible.

In this case, can we achieve the same result without moving the code ?

Reply:
No we cannot. This is how we did it for Kafka, vSphere, Istio, etc.